pkg/watchdog/pkg_test.go

```diff
@@ -4,8 +4,374 @@
 
 package watchdog
 
-import "testing"
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
 
-func TestTODO(t *testing.T) {
-	// TODO: Write a unit test.
+	"golang.org/x/sys/unix"
+)
+
+func getTestStruct() (*Watchdog, error) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		return nil, err
+	}
+	return &Watchdog{
+		File:               f,
+		unixSyscall:        nil,
+		unixIoctlGetUint32: nil,
+	}, nil
+}
+
+func TestOpen(t *testing.T) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		t.Error(err)
+	}
+	defer f.Close()
+
+	w, err := Open(f.Name())
+	if err != nil {
+		t.Error(err)
+	}
+	defer w.Close()
+}
+
+func TestOpenError(t *testing.T) {
+	_, err := Open("testfile")
+	if err == nil {
+		t.Error("Open succeeded but should have failed")
+	}
+}
+
+func TestSupport(t *testing.T) {
+	w, err := getTestStruct()
+	if err != nil {
+		t.Error("failed to construct test watchdog struct")
```

Comment:
Make sure to follow the got/want format. Here and everywhere else with a `t.Error`.